docs/src/main/asciidoc/kafka-streams.adoc

```diff
@@ -722,3 +722,226 @@ For more details about the health information, see the
 https://docs.spring.io/spring-boot/docs/current/reference/html/production-ready-endpoints.html#production-ready-health[Spring Boot Actuator documentation].
 
 NOTE: The status of the health indicator is `UP` if all the Kafka threads registered are in the `RUNNING` state.
+
+=== Functional Kafka Streams Applications
+
+With 2.2.0.RELEASE, Kafka Streams binder supports the ability to write applications by creating java.util.function.Function or java.util.consumer.Consumer beans.
+In this section, we will see the details of how the functional support work in the binder.
+The above `StreamListener` based model can be converted as below.
+
+[source]
+----
+@SpringBootApplication
+@EnableBinding(KafkaStreamsProcessor.class)
+public class WordCountProcessorApplication {
+
+	@Bean
+	public Function<KStream<?, String>, KStream<?, WordCount>> process() {
+		return input ->
+		        input
+                    .flatMapValues(value -> Arrays.asList(value.toLowerCase().split("\\W+")))
+                    .groupBy((key, value) -> value)
+                    .windowedBy(TimeWindows.of(5000))
+                    .count(Materialized.as("WordCounts-multi"))
+                    .toStream()
+                    .map((key, value) -> new KeyValue<>(null, new WordCount(key.key(), value, new Date(key.window().start()), new Date(key.window().end()))));
+    }
+
+	public static void main(String[] args) {
+		SpringApplication.run(WordCountProcessorApplication.class, args);
+	}
+----
+
+The input will be received from the input binding defined in the `KafkaStreamsProcessor` interface and the output will be sent to the output binding.
```

Comment:
The handler doesn't have any input argument, though. It is hard to relate to how and where the input binding is done. If the example is somehow converted to include an input argument, it'd then be self-explanatory in some ways. 

If the input binding is supposed to be treated as a black box, then the previous comment doesn't apply, but we can state that explicitly for readers.

Reply:
Oh wait, based on this signature `Function<KStream<?, String>, KStream<?, WordCount>>`, the input is a stream of strings and the output is a stream of `WordCount` objects. Maybe we can state that explicitly.